packages/web/src/shared/hooks/flash.tsx

```diff
@@ -0,0 +1,164 @@
+  }, []);
+
+  return (
+    <FlashContext.Provider value={{ error, success, remove, clear }}>
+      {children}
+      <div className="fixed top-0 w-full flex justify-center items-start z-10 pointer-events-none">
+        {messages.slice(0, 1).map((message) => {
+          return (
+            <Transition
+              key={message.id}
+              appear={message.state === 'unshown'}
+              show={message.state !== 'dismissed'}
+              className={`
+                mt-2 rounded shadow border font-bold flex items-stretch bg-white
+                pointer-events-auto
+                ${
+                  message.level === 'success'
+                    ? 'border-green-600'
+                    : 'border-red-600'
+                }
+              `}
+              enter="transition-opacity transition-transform duration-300"
+              enterFrom="opacity-0 -translate-y-4"
+              enterTo="opacity-100 translate-y-0"
+              leave="transition-opacity duration-75"
+              leaveFrom="opacity-100"
+              leaveTo="opacity-0"
+              afterEnter={() => {
+                setMessages((messages) =>
+                  messages.map((m, i) => ({ ...m, state: 'shown' }))
+                );
+              }}
+              afterLeave={() =>
+                setMessages((messages) =>
+                  messages.filter((m) => m.id !== message.id)
+                )
+              }
+            >
+              <div
+                className={`
+                w-9
+                ${message.level === 'success' ? 'bg-green-600' : 'bg-red-600'}
+              `}
+              >
+                <div className="flex justify-center items-center py-2 h-10">
+                  <Icon
+                    icon={
+                      message.level === 'success'
+                        ? 'check'
+                        : 'exclamation-triangle'
+                    }
+                    size="lg"
+                    className=" text-white"
+                  />
+                </div>
+              </div>
+              <div role="alert" className="py-2 px-3">
+                <span className="sr-only">
+                  {message.level === 'error' ? t('error') : t('success')}:
+                </span>
+                {message.message}
+              </div>
+              <button
+                type="button"
+                className="w-10 h-10 rounded focus:outline focus:outline-2 focus:outline-blue-600"
+                onClick={() => remove(message.id)}
+              >
+                <Icon icon="close" />
+                <span className="sr-only">{t('close')}</span>
+              </button>
+            </Transition>
+          );
+        })}
+      </div>
+    </FlashContext.Provider>
+  );
+}
+
+export function useFlash() {
+  const context = useContext(FlashContext);
+  if (!context) {
+    throw new Error('useFlash should be used within FlashContext component.');
+  }
+  return context;
+}
```

Comment:
In order to display a flash message, you can use this hook to get access to the functions in the flash provider